packages/react/src/__tests__/ReactJSXElementValidator-test.js

```diff
@@ -341,6 +341,32 @@ describe('ReactJSXElementValidator', () => {
     );
   });
 
+  it('does not warn when using getDefaultProps with createReactClass', () => {
+    // Set NODE_ENV to 'production' to verify that dev specific flags
+    // are not necessary
+    const ORIGINAL_ENV = process.env;
+    process.env = {
+      ...ORIGINAL_ENV,
+      NODE_ENV: 'production',
+    };
+    const createReactClass = require('create-react-class');
+    const ValidGetDefaultPropsComponent = createReactClass({
+      getDefaultProps: function() {
+        return {
+          prop: 'foo',
+        };
+      },
+      render: function() {
+        return <span>{this.props.prop}</span>;
+      },
+    });
+
+    spyOnDevAndProd(console, 'error');
+    ReactTestUtils.renderIntoDocument(<ValidGetDefaultPropsComponent />);
+    expect(console.error).not.toHaveBeenCalled();
+    process.env = ORIGINAL_ENV;
```

Comment:
Isn't it a problem that this will not be reverted to its original value if the test fails?

Reply:
Good call.  I pushed a commit to move this test to be under a `describe` block and ensured the `process.env` is reset in a `afterAll` block.  Thanks for the feedback!